File: calendarview/src/main/java/al/bruno/calendar/view/adapter/CustomViewHolder.java
package al.bruno.calendar.view.adapter;

import android.view.View;

import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;
/**
 * Created by bruno on 6/23/2017.
 */

public class CustomViewHolder<T, VM extends ViewDataBinding> extends RecyclerView.ViewHolder {
    private VM binding;
    private BindingInterface<T, VM> bindingInterface;

    public CustomViewHolder(View view, BindingInterface<T, VM> bindingInterface) {
        super(view);
        binding = DataBindingUtil.bind(view);
        this.bindingInterface = bindingInterface;
    }
    public void bindData(T model) {
        bindingInterface.bindData(model, binding);
        binding.executePendingBindings();
    }
}

File: genericadapter/src/main/java/al/bruno/adapter/holder/CustomViewHolder.kt
package al.bruno.adapter.holder

import al.bruno.adapter.BindingData
import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView

class CustomViewHolder<T, VM : ViewDataBinding>(itemView: View, private val bindingData: BindingData<T, VM>) : RecyclerView.ViewHolder(itemView) {
    private val binding: VM = DataBindingUtil.bind(itemView)!!
    fun bind(t:T) {
        bindingData.bindData(t, binding);
        binding.executePendingBindings()
    }
}